src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1938,8 +2007,85 @@ private void AddMeshData(string id, HelixRenderPackage rp, int index, int count,
 
             meshGeometry3D.Geometry = mesh;
             meshGeometry3D.Name = baseId;
+            meshGeometry3D.Tag = id;
+
+            if (intances != null)
+            {
+                meshGeometry3D.Instances = intances;
+            }
+        }
+
+        private void AddLineData(string id, HelixRenderPackage rp, int index, int count, bool drawDead, string baseId, List<Matrix> intances = null)
+        {
+            FastList<Vector3> lPositions;
+            FastList<Color4> lColors;
+            FastList<int> lIndices;
+
+            var l = rp.Lines;
+            if (index == 0 && count == rp.Lines.Positions.Count)
+            {
+                lPositions = l.Positions;
+                lColors = l.Colors;
+                lIndices = l.Indices;
+            }
+            else
+            {
+                lPositions = l.Positions.GetRange(index, count);
+                lColors = l.Colors.GetRange(index, count);
+                var firstIndicesIndex = l.Indices.IndexOf(index);
+                var indicesCount = l.Indices.IndexOf(index + count - 1) - firstIndicesIndex + 1;
+                lIndices = l.Indices.GetRange(firstIndicesIndex, indicesCount);
+            }
+
+            LineGeometryModel3D lineGeometry3D;
+
+            if (Element3DDictionary.ContainsKey(id))
+            {
+                lineGeometry3D = Element3DDictionary[id] as LineGeometryModel3D;
+            }
+            else
+            {
+                // If the package contains mesh vertices, then the lines represent the 
+                // edges of meshes. Draw them with a different thickness.
+                lineGeometry3D = CreateLineGeometryModel3D(rp, rp.MeshVertices.Any() ? 0.5 : 1.0);
```

Comment:
How were we doing this before? We weren't making a distinction between the thickness of lines and mesh edges?